usersession/client/client_test.go

```diff
@@ -448,12 +448,30 @@ func (s *clientSuite) TestServicesStopFailure(c *C) {
 }
 
 func (s *clientSuite) TestPendingRefreshNotification(c *C) {
+	n := 0
 	s.handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		n++
```

Comment:
This increment looks like it is racy. The client will be hitting both session agents in parallel, so if we're unlucky they'll both try to increment `n` at the same time. Maybe using a `sync.Mutex` or `atomic.AddInt32`?

Reply:
Nice catch, thanks, will do.

Reply:
Done.